aea/helpers/storage/backends/sqlite.py

```diff
@@ -0,0 +1,200 @@
+        super().__init__(uri)
+        parsed = urlparse(self._uri)
+        self._fname = parsed.netloc or parsed.path
+        self._connection: Optional[sqlite3.Connection] = None
+        self._loop: Optional[asyncio.AbstractEventLoop] = None
+        self._lock = threading.Lock()
+        self._executor = ThreadPoolExecutor(max_workers=1)
+
+    def _execute_sql_sync(self, query: str, args: Optional[List] = None) -> List[Tuple]:
+        """
+        Execute sql command and return results.
+
+        :param query: sql query string
+        :param args: optional argumets to set into sql query.
+
+        :return: List of tuples with sql records
+        """
+        if not self._connection:  # pragma: nocover
+            raise ValueError("Not connected")
+        with self._lock:
+            result = self._connection.execute(query, args or []).fetchall()
+            self._connection.commit()
+            return result
+
+    async def _executute_sql(self, query: str, args: Optional[List] = None):
+        """
+        Execute sql command and return results in async executor.
+
+        :param query: sql query string
+        :param args: optional argumets to set into sql query.
+
+        :return: List of tuples with sql records
+        """
+        if not self._loop:  # pragma: nocover
+            raise ValueError("Not connected")
+        return await self._loop.run_in_executor(
+            self._executor, self._execute_sql_sync, query, args
+        )
+
+    async def connect(self) -> None:
+        """Connect to backend."""
+        self._loop = asyncio.get_event_loop()
+        self._connection = await self._loop.run_in_executor(
+            self._executor, self._do_connect, self._fname
+        )
+
+    @staticmethod
+    def _do_connect(fname: str) -> sqlite3.Connection:
+        con = sqlite3.connect(fname)
+        if (
+            platform.system() == "Windows"
+            and sys.version_info.major == 3
+            and sys.version_info.minor < 9
+        ):  # pragma: nocover
+            con.enable_load_extension(True)
+            con.load_extension(
+                os.path.join(os.path.dirname(__file__), "binaries", "json1.dll")
+            )
+        return con
+
+    async def disconnect(self) -> None:
+        """Disconnect the backend."""
+        if not self._loop or not self._connection:  # pragma: nocover
+            raise ValueError("Not connected")
+        await self._loop.run_in_executor(self._executor, self._connection.close)
+        self._connection = None
+        self._loop = None
+
+    async def ensure_collection(self, collection_name: str) -> None:
+        """
+        Create collection if not exits.
+
+        :param collection_name: str.
+        :return: None
+        """
+        self._check_collection_name(collection_name)
+        sql = f"""CREATE TABLE IF NOT EXISTS {collection_name} (
+            object_id TEXT PRIMARY KEY,
+            object_body JSON1 NOT NULL)
+        """  # nosec
```

Comment:
use the appropriate way to avoid nosec

Reply:
https://security.openstack.org/guidelines/dg_parameterize-database-queries.html

Reply:
@DavidMinarsch 
not possible in my case, table names do not support substitution as a parameter.
check collection name performed before every query.